templates/al2023/provisioners/install-worker.sh

```diff
@@ -131,7 +131,31 @@ fi
 ###############################################################################
 
 sudo dnf install -y runc-${RUNC_VERSION}
-sudo dnf install -y containerd-${CONTAINERD_VERSION}
+
+# utility function for pulling rpms from an S3 bucket
+function rpm_install() {
+  local RPMS=($@)
+  echo "Pulling and installing local rpms from s3 bucket"
+  for RPM in "${RPMS[@]}"; do
+    aws s3 cp --region ${BINARY_BUCKET_REGION} s3://${BINARY_BUCKET_NAME}/rpms/${RPM} ${WORKING_DIR}/${RPM}
+    sudo yum localinstall -y ${WORKING_DIR}/${RPM}
+  done
+}
+
+# if k8sVersion >= 1.33, use the containerd 2.0 from s3 bucket
+# TO-DO: Currently using scratch build of containerd 2.0.4 for AL2023 in s3, change to use dnf install once it support
+if [[ "$CONTAINERD_VERSION" == 2.0* ]]; then
+  if sudo dnf install -y "containerd-2.0.*"; then
+    echo "Successfully installed containerd 2.0.* via dnf."
+  else
+    echo "Falling back to install containerd 2.0 rpm from S3."
+    arch=$(uname -m)
+    rpm_install "containerd-2.0.4-1.amzn2023.0.1.${arch}.rpm"
```

Comment:
this is a pretty common oneliner
```suggestion
    rpm_install "containerd-2.0.4-1.amzn2023.0.1.$(uname -m).rpm"
```